General integration improvements.

adam-the-hero opened the pull request.

nivek1612 committed:
Update manifest.json

Make HACS compliant

nivek1612 committed:
Update sensor.py

Wrong unit for usage. correct to Litres

nivek1612 committed:
Update sensor.py

Correct water meter duration unit of measure to minutes

nivek1612 committed:
Update sensor.py

Correct Device class for water meter usage in energy dashboard

nivek1612 committed:
Update sensor.py

Fix issues with syntax

nivek1612 committed:
Update sensor.py

correct syntax for  argument of type 'NoneType' is not iterable

nivek1612 committed:
Update __init__.py

Removed Deprecated code

nivek1612 committed:
Update __init__.py

deal with telemetry errors more elegantly

nivek1612 committed:
Update __init__.py

indentation error fixed

nivek1612 committed:
Update __init__.py

nivek1612 committed:
Update __init__.py

nivek1612 committed:
Update __init__.py

reverse later commits

adam-the-hero commented on custom_components/watergate/sensor.py:
```diff
@@ -90,8 +90,8 @@ async def async_setup_entry(
             WATER_METER_VOLUME_ENTITY_NAME,
-            UnitOfVolume.MILLILITERS,
-            SensorDeviceClass.VOLUME,
+            UnitOfVolume.LITERS,
```
@nivek1612 I think you cannot change that. We are providing the value in mililiter so you need to apply a conversion, right? And I cannot see it anywhere.

nivek1612 replied:
When it was millilitres the values being returned were always less than 1 so I figured you were returning litres here. Once I changed this to LITRES the values appeared to be of the right magnitude and similar to those reported in the mobile app. I wonder now if that means you are sending for example  7 LITRES as 7 millilitres rather than 7000 millilitres and that might be the reason for the discrepancy I see between the mobile app and the local API.

adam-the-hero replied:
That is interesting! We will figure it tomorrow.

nivek1612 replied:
cool

adam-the-hero replied:
So we checked that and we have ml in the telemetry reporting. I think you need to change that. But I've just schedule firmware upgrade so tomorrow morning it should work better.

nivek1612 replied:
Ok I'll recheck post the firmware update. Let me know when it is pushed out

nivek1612 replied:
@adam-the-hero just to check the value is a cumulative value. i.e. the total water volume since the integration started recording

adam-the-hero replied:
Yes it is total water volume registered by device. So basically it normal water meter.

adam-the-hero replied:
Will you change that? Or you want to do it after testing the new version of firmware?

nivek1612 replied:
Let me test the new version. I changed it back in the old version and it didn’t work as expected

adam-the-hero commented on custom_components/watergate/manifest.json:
```diff
@@ -1,11 +1,12 @@
   "documentation": "https://www.home-assistant.io/integrations/watergate",
   "iot_class": "local_push",
   "requirements": ["watergate-local-api==2024.2.1"],
+  "issue_tracker": "https://github.com/nivek1612/watergate-ha-custom-component/issues",
```
Can you just change it to original one?

nivek1612 replied:
Done

adam-the-hero reviewed (commented):
@nivek1612 Can you do some small updates?

nivek1612 committed:
Update manifest.json to point to main fork

nivek1612 committed:
Update sensor.py

No need to use the device class WATER this can be achieved via HA utility meter.
